sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -100,3 +104,17 @@
 // clang emits "warning:" in the message pragma output
 #define __SYCL_WARNING(msg) __pragma(message(msg))
 #endif // __GNUC__
+
+// Define __SYCL_UNROLL to add pragma/attribute unroll to a loop.
+#ifndef __SYCL_UNROLL
+#if defined(__INTEL_COMPILER) || defined(__INTEL_LLVM_COMPILER)
+#define __SYCL_UNROLL(x) _Pragma(__SYCL_STRINGIFY(unroll x))
+#elif defined(__clang__)
+#define __SYCL_UNROLL(x) _Pragma(__SYCL_STRINGIFY(unroll x))
+#elif (defined(__GNUC__) && __GNUC__ >= 8) ||                                  \
+    (defined(__GNUG__) && __GNUG__ >= 8)
+#define __SYCL_UNROLL(x) _Pragma(__SYCL_STRINGIFY(GCC unroll x))
+#else
+#define __SYCL_UNROLL(x)
+#endif // compiler switch
+#endif // __SYCL_UNROLL
```

Comment:
While looking at this I wonder why just using a metaprogrammed loop instead is not better than this pragmatic spaghetti plate.

Reply:
It also works with one-iteration loops, unlike this pragma-based solution: https://github.com/intel/llvm/pull/6560